service/src/main/java/bio/terra/workspace/common/utils/FlightUtils.java

```diff
@@ -164,4 +169,27 @@ public static <T> T getRequired(FlightMap flightMap, String key, Class<T> tClass
     }
     return value;
   }
+
```

Comment:
optional - I don't think this is needed. The natural place for this is next to Stairway.waitForFlight(). PF-1706 mentions 20 seconds. If you can get it down to 1 sec, that seems good enough.

Did you test this PR? How long did cloning take?

Reply:
One second is too long, because it sleeps first before checking the flight state. The flight only takes 35-50 ms. We want all of the reference clones to take place virtually instantaneously b/c we don't want another spinner in the UI.
![Screen Shot 2022-06-09 at 12 30 04 PM](https://user-images.githubusercontent.com/53479492/172897883-63f45e6a-74ef-4a7b-a751-8bc5ebc41c21.png)

Reply:
I didn't want to poll too frequently in the case where it takes longer, so I like the fact that it spreads out over time.

I'll put this in Stairway after giving it some run time in WSM.